cmake/FindFCGI.cmake

```diff
@@ -0,0 +1,74 @@
+# Created by Viacheslav Greshilov http://github.com/greshilov
+
+file(GLOB _BREW_FCGI "/usr/local/Cellar/fcgi/*")
```

Comment:
1. Does it work if there are several versions in the directory? GLOB should return a list of files/directories.
2. Does it select the highest available version?

As far as I remember, brew does not delete old package versions by default, only if you call `brew cleanup`.
Probably, something like `FOREACH(_BREW_FCGI_VERSION ${_BREW_FCGI})` is required with some additional logic to select the highest available version.

Reply:
By default ```find_path``` and ```find_library``` cmake functions try to find files in default system directories at first. If I'm not mistaken brew always create link in /usr/local/lib... e.t.c for last stable version of package after installing. For example cmake output on my system is: 
```
-- Found FCGI: /usr/local/lib/libfcgi.dylib  
-- Found FCGI++: /usr/local/lib/libfcgi++.dylib
```
This means that he had not even reached ```_BREW_FCGI``` variable during the search in source paths.

Anyway we can do something like this:
```
file(GLOB _BREW_FCGI "/usr/local/Cellar/fcgi/*")
list(SORT _BREW_FCGI)
list(REVERSE _BREW_FCGI)
```
If cmake not found *.h and lib files in system dirs it will iterate paths from variable ```_FCGI_SOURCE_PATHS``` one by one. ```list(REVERSE..``` changes order of this path  sequence from higher version to lower.

Reply:
I've created copy of fastcgi dir and call it 2.5.0.
```
ls /usr/local/Cellar/fcgi/ 
# Output:
# 2.4.0	2.5.0
```
Code example:
```
file(GLOB _BREW_FCGI "/usr/local/Cellar/fcgi/*")
list(SORT _BREW_FCGI)
list(REVERSE _BREW_FCGI)

set(
  _FCGI_SOURCE_PATHS
  ${_BREW_FCGI}
  /usr/
  /usr/local/
)

find_path(
  FCGI_INCLUDE_DIR
  NAMES fcgio.h fcgiapp.h
  PATHS ${_FCGI_SOURCE_PATHS}
  PATH_SUFFIXES include
  NO_DEFAULT_PATH
)
message(${FCGI_INCLUDE_DIR})
# Output:
# usr/local/Cellar/fcgi/2.5.0/include
```

Reply:
If brew always creates symlinks, the best solution would be to remove this location at all :) Why should we manually care about versions if brew is already doing it for us?

Reply:
For most cases, absolutely yes. But I'm not sure about the situation when specific version is installed. 
```
# Imagine the world in which more than one fcgi package is available on brew :)
brew install fcgi@2.5.1
```
In this case the symbolic link will not be made automatically. And code with _BREW_FCGI will did a trick! Anyway the last word belongs to you.

Reply:
Users of this code will not play with versions. They just `brew install fcgi` and run cmake.
Good to know that it's easy to handle situations like this in cmake. Let's do this: leave version-looking code commented for the future generations.

Reply:
Done!